libs/zwin32/src/dds_loader.zig

```diff
@@ -0,0 +1,867 @@
+            .@"packed" = true,
+            .planar = false,
+            .bpe = 4,
+        },
+
+        .Y210, .Y216 => FormatData{
+            .bc = false,
+            .@"packed" = true,
+            .planar = false,
+            .bpe = 8,
+        },
+
+        .NV12, .@"420_OPAQUE" => blk: {
+            // Requires a height alignment of 2.
+            // return E_INVALIDARG;
+            assert(height % 2 == 0);
+            break :blk FormatData{
+                .bc = false,
+                .@"packed" = false,
+                .planar = true,
+                .bpe = 2,
+            };
+        },
+
+        .P208 => FormatData{
+            .bc = false,
+            .@"packed" = false,
+            .planar = true,
+            .bpe = 2,
+        },
+
+        .P010, .P016 => blk: {
+            // Requires a height alignment of 2.
+            // return E_INVALIDARG;
+            assert(height % 2 == 0);
+            break :blk FormatData{
+                .bc = false,
+                .@"packed" = false,
+                .planar = true,
+                .bpe = 4,
+            };
+        },
+
+        else => FormatData{
+            .bc = false,
+            .@"packed" = false,
+            .planar = false,
+            .bpe = 0,
+        },
+    };
+
+    if (format_data.bc) {
+        var num_blocks_wide: u64 = 0;
+        if (width > 0) {
+            num_blocks_wide = std.math.max(1, @divTrunc(@intCast(u64, width) + 3, 4));
+        }
+        var num_blocks_high: u64 = 0;
+        if (height > 0) {
+            num_blocks_high = std.math.max(1, @divTrunc(@intCast(u64, height) + 3, 4));
+        }
+        row_bytes = num_blocks_wide * format_data.bpe;
+        num_rows = num_blocks_high;
+        num_bytes = row_bytes * num_blocks_high;
+    } else if (format_data.@"packed") {
+        row_bytes = ((@intCast(u64, width) + 1) >> 1) * format_data.bpe;
+        num_rows = @intCast(u64, height);
+        num_bytes = row_bytes * height;
+    } else if (format == .NV11) {
+        row_bytes = ((@intCast(u64, width) + 3) >> 2) * 4;
+        num_rows = @intCast(u64, height) * 2; // Direct3D makes this simplifying assumption, although it is larger than the 4:1:1 data
+        num_bytes = row_bytes * num_rows;
+    } else if (format_data.planar) {
+        row_bytes = ((@intCast(u64, width) + 1) >> 1) * format_data.bpe;
+        num_bytes = (row_bytes * @intCast(u64, height)) + ((row_bytes * @intCast(u64, height) + 1) >> 1);
+        num_rows = height + ((@intCast(u64, height) + 1) >> 1);
+    } else {
+        const bpp = bitsPerPixel(format);
+        assert(bpp > 0);
+
+        row_bytes = @divFloor(@intCast(u64, width) * bpp + 7, 8); // round up to nearest byte
+        num_rows = @intCast(u64, height);
+        num_bytes = row_bytes * height;
+    }
+
+    return .{
+        .num_bytes = num_bytes,
+        .row_bytes = row_bytes,
+        .num_rows = num_rows,
+    };
+}
+
+inline fn bitsPerPixel(format: dxgi.FORMAT) usize {
```

Comment:
We have `dxgi.Format.pixelSizeInBytes()` method, could we use it instead?

Reply:
I don't know what to return though for BC4_ formats as they use 4 bits per pixel 🤔

Reply:
Right, I think we should change it to return pixel size in bits. Should be easy because we use this function in one place only.

Reply:
On it